app/src/main/java/com/jerboa/feat/ModActions.kt

```diff
@@ -12,29 +16,25 @@ import java.time.temporal.ChronoUnit
 fun canMod(
     creatorId: PersonId,
     admins: List<PersonView>?,
-    moderators: List<CommunityModeratorView>?,
-    myId: PersonId?,
+    moderators: List<PersonId>?,
+    myId: PersonId,
     onSelf: Boolean = false,
 ): Boolean {
-    return if (myId !== null) {
-        // You can do moderator actions only on the mods added after you.
-        val adminIds = admins?.map { a -> a.person.id }.orEmpty()
-        val modIds = moderators?.map { m -> m.moderator.id }.orEmpty()
-
-        val adminsThenMods = adminIds.toMutableList()
-        adminsThenMods.addAll(modIds)
-
-        val myIndex = adminsThenMods.indexOf(myId)
-        if (myIndex == -1) {
-            false
-        } else {
-            // onSelf +1 on mod actions not for yourself, IE ban, remove, etc
-            val subList = adminsThenMods.subList(0, myIndex.plus(if (onSelf) 0 else 1))
-
-            !subList.contains(creatorId)
-        }
-    } else {
+    // You can do moderator actions only on the mods added after you.
```

Comment:
Does the moderator hierarchy not apply to admins? i.e. Any admin can moderate any admin?

Reply:
It does apply to other admins. Check L28 and L35, it adds admins, then mods for the ordered hierarchy.

Reply:
It's not about the code, but rather does Lemmy also have the hierarchy for admin

Reply:
Yep admins also have a hierarchy.